src/components/TodoForm/TodoForm.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import './TodoForm.css';
+
+export class TodoForm extends React.PureComponent {
+  state = {
+    enteredValue: '',
+    selectedUser: '',
+    showValueError: false,
+    showUserError: false,
+    userID: 3,
+  }
+
+  changeHandler = (event) => {
+    const { name, value } = event.target;
+
+    this.setState({ [name]: value });
+  }
+
+  addHandler = () => {
+    const { addTodo } = this.props;
+    const { enteredValue, selectedUser, userID } = this.state;
+
+    if (!enteredValue.trim() || !selectedUser) {
+      if (!enteredValue.trim()) {
+        this.setState({ showValueError: true });
+        setTimeout(() => (this.setState({ showValueError: false })), 2000);
+      }
+
+      if (!selectedUser) {
+        this.setState({ showUserError: true });
+        setTimeout(() => (this.setState({ showUserError: false })), 2000);
+      }
+
+      return;
+    }
+
+    this.setState(prevState => ({
+      enteredValue: '',
+      selectedUser: '',
+      showValueError: false,
+      showUserError: false,
+      userID: prevState.userID + 1,
+    }));
+
+    addTodo({
+      userId: userID,
+      id: userID,
+      title: enteredValue,
+      completed: false,
+      user: {
+        name: selectedUser,
+      },
+    });
+  }
+
+  render() {
+    const { userList } = this.props;
+    const {
+      enteredValue,
+      selectedUser,
+      showValueError,
+      showUserError,
+    } = this.state;
+
+    return (
+      <div>
+        <form className="form" onSubmit={event => event.preventDefault()}>
```

Comment:
add here to onSubmit handleSubmit func, move event.preventDefault inside this func